kratos/modeler/create_entities_from_geometries_modeler.h

```diff
@@ -0,0 +1,183 @@
+
+// Project includes
+#include "containers/model.h"
+#include "includes/define.h"
+#include "includes/define_registry.h"
+#include "includes/kratos_components.h"
+#include "modeler/modeler.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class CreateEntitiesFromGeometriesModeler
+ * @ingroup KratosCore
+ * @brief Modeler to create element/condition entities from geometries
+ * @details Given a list of pairs of the type submodelpart name and entity (element/condition) name,
+ * this modeler creates the corresponding entities from the submodelpart geometries.
+ * @author Ruben Zorrilla
+ */
+class KRATOS_API(KRATOS_CORE) CreateEntitiesFromGeometriesModeler
+    : public Modeler
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Modeler
+    KRATOS_CLASS_POINTER_DEFINITION(CreateEntitiesFromGeometriesModeler);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor
+    CreateEntitiesFromGeometriesModeler() : Modeler() {};
+
+    /// Constructor with Model
+    CreateEntitiesFromGeometriesModeler(
+        Model &rModel,
+        Parameters Settings)
+        : Modeler(rModel, Settings)
+    {
+        mpModel = &rModel;
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+    }
+
+    /// Destructor.
+    ~CreateEntitiesFromGeometriesModeler() = default;
+
+    /// Creates the CreateEntitiesFromGeometriesModeler Pointer
+    Modeler::Pointer Create(
+        Model& rModel,
+        const Parameters Settings) const override
+    {
+        return Kratos::make_shared<CreateEntitiesFromGeometriesModeler>(rModel, Settings);
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Convert the geometry model or import analysis suitable models.
+    void SetupModelPart() override;
+
+    /// This method provides the defaults parameters to avoid conflicts between the different constructors
+    const Parameters GetDefaultParameters() const override
+    {
+        const Parameters default_parameters = Parameters(R"({
+            "echo_level" : 0,
+            "elements_list" : [],
+            "conditions_list" : []
+        })");
+        return default_parameters;
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "CreateEntitiesFromGeometriesModeler";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    KRATOS_REGISTRY_ADD_PROTOTYPE("Modelers.KratosMultiphysics", CreateEntitiesFromGeometriesModeler)
```

Comment:
Caution! Registry KRATOS_REGISTRY_ADD_PROTOTYPE has 3 args after merging master into this branch

Reply:
Solved.